packages/langium/src/parser/token-builder.ts

```diff
@@ -38,11 +38,6 @@ export class DefaultTokenBuilder implements TokenBuilder {
                 tokens.push(terminalToken);
             }
         });
-
-        //reminder: EOF should always be the last token, because it is very unlikely that it will be matched within the input
-        if (reachableRules.some(r => streamAst(r.definition).some(isEndOfFile))) {
-            tokens.push(EOF);
-        }
```

Comment:
I think this is worth a comment, maybe with a reference to a statement in chevrotain docs or code.
Otherwise, nobody except you will have a clue why it is like it is.